README.md

```diff
@@ -1,13 +1,203 @@
+| `1.2.3-beta.2.1` > `1.2.3-beta.1` | If all of the preceding identifiers are equal, integers must be compared the normal way. |
+| `1.2.3-beta.2` > `1.2.3-beta.1` | If all of the preceding identifiers are equal, integers must be compared the normal way. |
+| `1.2.3-beta` > `1.2.3-1` | Where strings and integers must be compared, strings have a higher precedence than integers. |
+| `1.2.3-beta+abcd` = `1.2.3-beta` | Build-metadata is not to be used when comparing. |
+| `1.2.3-beta+sha.0beef` = `1.2.3-beta+sha.80081` | Build-metadata is not to be used when comparing. |
+
+## Library Usage
+#### Creating the object directly:
+Imports the library, and creates a `SemanticVersion` object directly.
 ```toit
-import semver
+import semver show *
 
 main:
-  a := "1.0.0"
-  b := "1.0.0-beta.1"
-  print (semver.compare a b)  // => 1
+  // Instantiation by direct creation
+  semver-a := SemanticVersion 1 2 3
+
+  // Prints 1.2.3
+  print "$semver-a"
+
+  // Instantiation by direct creation, without 'minor'
+  semver-b := SemanticVersion 1 2
+
+  // Prints 1.2.0
+  print "$semver-b"
+
+  // Instantiation by direct creation, without 'minor' or 'patch'
+  semver-c := SemanticVersion 1
+
+  // Prints 1.0.0
+  print "$semver-c"
+```
+#### Directly creating including pre-release:
+Pre-release and build-metadata can also be specified directly.  These are held
+as a list and must be specified as a list, even if only one element:
+```toit
+  // (Contiues from previous example)
+
+  // Direct instantiation.
+  semver-d := SemanticVersion 1 0 0 ["alpha","1"] ["sha",23132]
+
+  // Prints 1.0.0-alpha.1+sha.23132
+  print "$semver-d"
+```
+#### Immutability:
+Since the object is immutable, editing one of the fields after creation is not
+possible.
+```toit
+  // (Contiues from previous examples)
+
+  // Direct instantiation including pre-release.
+  semver-e := SemanticVersion 1 2 3 ["alpha",1]
+
+  // Prints 2
+  print "$(semver-e.minor)"
+
+  // Fails/throws
+  //semver-e.minor = 5
+```
+#### Object instantiation by string parsing:
+The library parses strings into a `SemanticVersion` object, which has methods
+and functions.  Comparison operators are shown in the example below.
+```toit
+  // (Contiues from previous examples)
+
+  // strings
+  string-f := "1.0.0"
+  string-g := "1.0.0-beta.1"
+
+  // parse strings into SemanticVersion objects
+  semver-f := SemanticVersion.parse string-f
+  semver-g := SemanticVersion.parse string-g
+
+  // compare two objects: prints "f is later than g."
+  if semver-f > semver-g:
+    print "f is later than g."
+  else:
+    print "g is later than f."
+
+  // compare two objects: prints "f and g are different."
+  if semver-f == semver-g:
+    print "a and b are the same."
+  else:
+    print "a and b are different."
+
+  // compare two objects: prints "f and f are the same."
+  if semver-f == semver-f:
+    print "f and f are the same."
+  else:
+    print "f and f are different."
+
```

Comment:
Not needed.
```suggestion
```

Consider adding examples for the yellow strings, if we accept them with specific flags.
In that case, it's also interesting to know whether a stringification/to-string returns the original string or a correct one.
So does `(SemanticVersion.parse "1.1").stringify` return "1.1" or "1.1.0"?